core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/rds/RdsUpgradeService.java

```diff
@@ -86,7 +104,16 @@ private void alreadyOnLatestAnswer(TargetMajorVersion targetMajorVersion) {
         throw new BadRequestException(message);
     }
 
-    private RdsUpgradeV4Response validateAndTrigger(StackView stack, TargetMajorVersion targetMajorVersion, String accountId) {
+    private RdsUpgradeV4Response validateAndTrigger(NameOrCrn nameOrCrn, StackView stack, TargetMajorVersion targetMajorVersion, String accountId) {
+        validate(nameOrCrn, stack, targetMajorVersion, accountId);
+        LOGGER.info("External database for stack {} will be upgraded to version {}", stack.getName(), targetMajorVersion.getMajorVersion());
+        return triggerRdsUpgradeFlow(stack, targetMajorVersion);
+    }
+
+    private void validate(NameOrCrn nameOrCrn, StackView stack, TargetMajorVersion targetMajorVersion, String accountId) {
+        if (getCurrentRdsVersion(nameOrCrn).equals(targetMajorVersion.getMajorVersion())) {
+            alreadyOnLatestAnswer(targetMajorVersion);
+        }
```

Comment:
I would restructure this method to either:
```
if(check1()) {
    throwAnswer()
} else if (check2()){ 
   throwAnswers2())
}
```

Or the one I would prefer the most:
```
validateRdsIsnotUpgraded();
validateRuntimeEligibleForUpgrade();
validateAttachedDatahubsAreNotRunning();
```
And within each validateXXX call there would be an exception thrown if validation failed.